File: SFA-Client/src/app/_models/orderItemBatch.ts
export class OrderItemBatch {
  id: number;
  orderId: number;
  itemBatchId: string;
  orderItemBatchCode: number;
  name: string;
  itemCount: number;
  sellingToShopOwnerAmount: number;
  shopOwnerProfitAmount: number;
  companyProfitAmount: number;
  companyDiscountRate: number;
  shopOwnerDiscountRate: number;
  isSpecialDiscountHave: boolean;
  customerFreeIssueQuantity: number;
  shopOwnerFreeIssueQuantity: number;

  constructor(orderItemBatch) {
    this.id = orderItemBatch.id;
    this.orderId = orderItemBatch.orderId;
    this.itemBatchId = orderItemBatch.itemBatchId;
    this.orderItemBatchCode = orderItemBatch.orderItemBatchCode;
    this.name = orderItemBatch.name;
    this.itemCount = orderItemBatch.itemCount;
    this.sellingToShopOwnerAmount = orderItemBatch.sellingToShopOwnerAmount;
    this.shopOwnerProfitAmount = orderItemBatch.shopOwnerProfitAmount;
    this.companyProfitAmount = orderItemBatch.companyProfitAmount;
    this.companyDiscountRate = orderItemBatch.companyDiscountRate;
    this.shopOwnerDiscountRate = orderItemBatch.shopOwnerDiscountRate;
    this.isSpecialDiscountHave = orderItemBatch.isSpecialDiscountHave;
    this.customerFreeIssueQuantity = orderItemBatch.customerFreeIssueQuantity;
    this.shopOwnerFreeIssueQuantity = orderItemBatch.shopOwnerFreeIssueQuantity;
  }
}

File: SFA-API/Domain/OrderItemBatch.cs
namespace Domain
{
    public class OrderItemBatch
    {
        public int Id { get; set; }
        public int OrderId { get; set; }
        public int ItemBatchId { get; set; }
        public string OrderItemBatchCode { get; set; }
        public string Name { get; set; }
        public int ItemCount { get; set; }
        public decimal SellingToShopOwnerAmount { get; set; }
        public decimal ShopOwnerProfitAmount { get; set; }
        public decimal CompanyProfitAmount { get; set; }
        public decimal CompanyDiscountRate { get; set; }
        public decimal ShopOwnerDiscountRate { get; set; }
        public bool IsSpecialDiscountHave { get; set; }
        public int CustomerFreeIssueQuantity { get; set; }
        public int ShopOwnerFreeIssueQuantity { get; set; }
        public virtual ItemBatch ItemBatch { get; set; }
        public virtual Order Order { get; set; }
    }
}